sql/parser/builtins.go

```diff
@@ -72,6 +74,55 @@ var builtins = map[string]builtin{
 	"upper": stringBuiltin(func(s string) (Datum, error) {
 		return DString(strings.ToUpper(s)), nil
 	}),
+
+	"substr": {
+		nArgs: 3,
+		fn: func(args DTuple) (Datum, error) {
```

Comment:
The syntax is handled in the grammar but we'll need to fill out the semantic actions. Should be straightforward to do. I think we should just translate that specific syntax into the syntax used here.